userguide/payment/includes/payment-states.adoc

```diff
@@ -55,29 +55,30 @@ Note that the first 3 cases are normal cases but the last 2 are errors that are
+The polling delay is configured via the  `org.killbill.payment.janitor.unknown.retries` property for UNKNOWN transactions (default `5m,1h,1d,1d,1d,1d,1d`) and the `org.killbill.payment.janitor.pending.retries` property for PENDING transactions (default `1h, 1d`). These properties can be configured as explained in the https://docs.killbill.io/latest/userguide_configuration.html[_Configuration Guide_]. Companies relying on PENDING for offline payment methods (ACH, Boleto, etc.) should increase the number of attempts, as such payments can be slow (they can take multiple days to execute).
 
 ==== On-the-fly Janitor
 
 The on-the-fly Janitor is invoked:
 
-* Whenever a payment (or set of payments) is retrieved via API and the flag `withPluginInfo` is set. In this case, a GET operation might result in data being written to disk (but it's not really new state, it's just existing state being updated that was incomplete).
+* Whenever a payment (or set of payments) is https://github.com/killbill/killbill-api/blob/4ae1c343a593de937415e21feecb9f5405037fa3/src/main/java/org/killbill/billing/payment/api/PaymentApi.java#L439[retrieved via API] and the flag `withPluginInfo` is set. In this case, a GET operation might result in data being written to disk (but it's not really new state, it's just existing state being updated that was incomplete).
 * Whenever an operation is performed on an existing payment (either initiated internally by Kill Bill or externally via our payment APIs): the Janitor is invoked first to get the latest state to ensure the internal state machine prevents disallowed transitions in case the state couldn't be fixed (or if it's already in a final state). For example, if an authorization state is currently UNKNOWN and you attempt to capture it, the Janitor will ask the plugin first to fix the state and either let the capture go through (if the authorization is fixed as SUCCESS) or disallow it (if the authorization couldn't be fixed or was unsuccessful).
 
 Note however that the `notifyPendingPaymentOfStateChanged` operation will not invoke the Janitor (as this API is often called from payment plugins, we don't want the state to be fixed twice).
 
 ==== Janitor and Control Plugins
 
-When the Janitor fixes a payment, it will re-run the Payment Control state machine (i.e. go through Payment Control plugins) if needed. So for instance, invoice payments that were created as part of a RECURRING invoicing may end up being fixed automatically by the Janitor, and the balance associated with the invoice would then be correctly reflected.
+When the Janitor fixes a payment, it will re-run the Payment Control state machine (i.e. go through Payment Control plugins) if needed. So for instance, invoice payments that were created as part of a RECURRING invoice may end up being fixed automatically by the Janitor, and the balance associated with the invoice would then be correctly reflected.
```

Comment:
```suggestion
When the Janitor fixes a payment, it will re-run the Payment Control state machine (i.e. go through Payment Control plugins) if needed. For instance, invoice payments that were created as part of a RECURRING invoice may end up being fixed automatically by the Janitor, and the balance associated with the invoice would then be correctly reflected.
```